channels_redis/core.py

```diff
@@ -208,6 +208,9 @@ def _setup_encryption(self, symmetric_encryption_keys):
                     await self.receive_buffer[channel].put(message)
             else:
                 await self.receive_buffer[real_channel].put(message)
+            # this sleep makes sure that the redis queue is not popped before the
+            # previous message was processed and returned back to the caller
+            await  asyncio.sleep(0.0001)
```

Comment:
i am not happy with this, but i couldn't find any other solution to this. as when i used connection pools , the last test case which was creating an issue:

Test Case  No 16 :
get a new channel - > add channel to group -> send four messages -> one by one call receive

PROBLEM : Alternate messages were getting skipped and following flow of events is being followed
  
receive -> start receive_loop -> wait for pop result -> pop result_1 -> pop result_2 -> process and return result 1 from receive -> CANCEL receive_loop -> call receive again -> pop result_3 -> process and return result_3

result_2 was being popped from the queue and the task was then getting cancelled. let me know if you come up with a better solution.

Reply:
This sleep schedules the receive coroutine before again awaiting the receive_single coroutine which in turns pops the redis queue. now the receive_loop was cancelled before even the message was popped from redis.